### PR TITLE
iptables: filter by IP address (and range)

### PR DESCRIPTION
**Pull request by kevinGC**
Enables commands such as:
$ iptables -A INPUT -d 127.0.0.1 -j ACCEPT
$ iptables -t nat -A PREROUTING ! -d 127.0.0.1 -j REDIRECT

Also adds a bunch of REDIRECT+destination tests.

### CODE REVIEW

File: test/iptables/filter_output.go [hunk 1]
```diff
@@ -85,3 +87,57 @@ func (FilterOutputDropTCPSrcPort) LocalAction(ip net.IP) error {
...
+	return sendUDPLoop(ip, acceptPort, sendloopDuration)
+}
+
+// LocalAction implements TestCase.LocalAction.
+func (FilterOutputDestination) LocalAction(ip net.IP) error {
+	return listenUDP(acceptPort, sendloopDuration)
+}
+
+// FilterOutputInvertDestination tests that we can selectively allow packets
+// not headed for a particular destionation.
```
Comment (nybidari) [inline test/iptables/filter_output.go]: Nit: destination

File: test/iptables/iptables_util.go [hunk 2]
```diff
@@ -46,8 +47,17 @@ func tableCmd(table string, args []string) error {
 
 // filterTableRules is like filterTable, but runs multiple iptables commands.
 func filterTableRules(argsList [][]string) error {
+	return tableRules("filter", argsList)
+}
+
+// filterTableRules is like filterTable, but runs multiple iptables commands.
```
Comment (nybidari) [inline test/iptables/iptables_util.go]: Nit: natTableRules

File: test/iptables/nat.go [hunk 3]
```diff
@@ -78,3 +88,192 @@ func (NATDropUDP) ContainerAction(ip net.IP) error {
...
+type NATOutDontRedirectIP struct{}
+
+// Name implements TestCase.Name.
+func (NATOutDontRedirectIP) Name() string {
+	return "NATOutDontRedirectIP"
+}
+
+// ContainerAction implements TestCase.ContainerAction.
+func (NATOutDontRedirectIP) ContainerAction(ip net.IP) error {
+	if err := natTable("-A", "OUTPUT", "-d", "127.0.0.1", "-p", "udp", "-j", "REDIRECT", "--to-port", fmt.Sprintf("%d", dropPort)); err != nil {
```
Comment (nybidari) [inline test/iptables/nat.go]: Use localIP instead of "127.0.0.1"

File: test/iptables/filter_output.go (hunk 1 above)
Comment (kevinGC) [reply test/iptables/filter_output.go]: Done.

File: test/iptables/iptables_util.go (hunk 2 above)
Comment (kevinGC) [reply test/iptables/iptables_util.go]: Done.

File: test/iptables/nat.go (hunk 3 above)
Comment (kevinGC) [reply test/iptables/nat.go]: Done,

**Pull request merged**